### PR TITLE
[DRAFT] Allow ponyfilling window.open

### LINKED ISSUE #368: Allow ponyfilling window.open
Right now we call `window.open(url)` for card actions. The default in most browsers is to open the url in a new tab, but some might like to open it in a new window, etc.

My proposal is to add `urlTarget?:string` and `urlTitle?:string` to `ChatProps`. If supplied we'd pass them as the second and third parameter to `window.open`.

### PR DESCRIPTION
**Pull request by justinwilaby**
> Fix #368.

### CODE REVIEW

File: packages/component/src/Composer.js [hunk 1]
```diff
@@ -96,9 +96,8 @@ function createCardActionLogic({ directLine, dispatch }) {
         case 'playAudio':
         case 'playVideo':
         case 'showImage':
-          // TODO: [P3] We should support ponyfill for window.open
-          //       This is as-of v3
-          window.open(value);
+          const factory = this.props.externalUrlHandlerFactory({ value });
```
Comment (justinwilaby) [inline packages/component/src/Composer.js]: @compulim - Let me know what you think should be included in the factory's function signature.

Comment (compulim) [reply packages/component/src/Composer.js]: How about?

`windowOpenPonyfillFactory({ activity, attachment, cardAction }) => (url: string, windowName: string, windowFeatures: string) => void`

For the object returned from the ponyfill, we can assume we are not acting on it, until we need it. So I am leaving it `void` right now.

**Comment by cwhitten** [comment]
@compulim @justinwilaby where are we at on this?

**Comment by compulim** [comment]
Will fix in #1704. Closing this for now.

**Pull request closed without merging**